app/test/state_sync_test.go

```diff
@@ -0,0 +1,68 @@
+package app_test
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/celestiaorg/celestia-app/v3/app"
+	"github.com/celestiaorg/celestia-app/v3/app/encoding"
+	"github.com/celestiaorg/celestia-app/v3/test/util"
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/snapshots"
+	snapshottypes "github.com/cosmos/cosmos-sdk/snapshots/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	abcitypes "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/libs/log"
+	dbm "github.com/tendermint/tm-db"
+)
+
+func TestOfferSnapshot(t *testing.T) {
+	t.Run("v1", func(t *testing.T) {
+		testApp := createTestApp(t)
+		request := createRequest()
+		request.AppVersion = 1
+		resp := testApp.OfferSnapshot(request)
+		assert.Equal(t, abcitypes.ResponseOfferSnapshot{Result: abcitypes.ResponseOfferSnapshot_ACCEPT}, resp)
+	})
+	t.Run("v2", func(t *testing.T) {
+		testApp := createTestApp(t)
+		request := createRequest()
+		request.AppVersion = 2
+		resp := testApp.OfferSnapshot(request)
+		assert.Equal(t, abcitypes.ResponseOfferSnapshot{Result: abcitypes.ResponseOfferSnapshot_ACCEPT}, resp)
```

Comment:
Nice tests! Can we also add a test for when the app version is not supported? like v3 in this case?

Reply:
Can you also shed light on the difference between the test in state_sync_test.go (`TestOfferSnapshot`) and the ones added to the in app_test.go? It is not immediately clear how they differ. Thanks!

Reply:
> Nice tests! Can we also add a test for when the app version is not supported? like v3 in this case?

v3 is supported because this PR targets main. 

> Can you also shed light on the difference between the test in state_sync_test.go (TestOfferSnapshot) and the ones added to the in app_test.go? It is not immediately clear how they differ. Thanks!

Great question, I had the same realization because I added unit tests for v1, v2, v3, v4 in the other file but missed this one. The tests look the same so will delete one of them.

Reply:
Fixed in https://github.com/celestiaorg/celestia-app/pull/3936/commits/ad1f3b3c35c592ff48bc8a24189ba7fd27e5c858